[2024-05-06T07:40:59.990+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2024-05-06T07:41:00.118+0000] {taskinstance.py:2073} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: tweets_dag_v4.execute_python_script scheduled__2024-04-30T00:00:00+00:00 [queued]>
[2024-05-06T07:41:00.141+0000] {taskinstance.py:2073} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: tweets_dag_v4.execute_python_script scheduled__2024-04-30T00:00:00+00:00 [queued]>
[2024-05-06T07:41:00.143+0000] {taskinstance.py:2303} INFO - Starting attempt 3 of 8
[2024-05-06T07:41:00.188+0000] {taskinstance.py:2327} INFO - Executing <Task(PythonOperator): execute_python_script> on 2024-04-30 00:00:00+00:00
[2024-05-06T07:41:00.220+0000] {logging_mixin.py:188} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:61 DeprecationWarning: This process (pid=504) is multi-threaded, use of fork() may lead to deadlocks in the child.
[2024-05-06T07:41:00.226+0000] {standard_task_runner.py:63} INFO - Started process 682 to run task
[2024-05-06T07:41:00.234+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'tweets_dag_v4', 'execute_python_script', 'scheduled__2024-04-30T00:00:00+00:00', '--job-id', '159', '--raw', '--subdir', 'DAGS_FOLDER/twitter_data_dag.py', '--cfg-path', '/tmp/tmpc8akq9h9']
[2024-05-06T07:41:00.241+0000] {standard_task_runner.py:91} INFO - Job 159: Subtask execute_python_script
[2024-05-06T07:41:00.420+0000] {task_command.py:426} INFO - Running <TaskInstance: tweets_dag_v4.execute_python_script scheduled__2024-04-30T00:00:00+00:00 [running]> on host bc28b2896df0
[2024-05-06T07:41:00.704+0000] {taskinstance.py:2644} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='stan' AIRFLOW_CTX_DAG_ID='tweets_dag_v4' AIRFLOW_CTX_TASK_ID='execute_python_script' AIRFLOW_CTX_EXECUTION_DATE='2024-04-30T00:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='3' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2024-04-30T00:00:00+00:00'
[2024-05-06T07:41:00.708+0000] {taskinstance.py:430} INFO - ::endgroup::
[2024-05-06T07:46:16.100+0000] {local_task_job_runner.py:310} WARNING - State of this instance has been externally set to None. Terminating instance.
[2024-05-06T07:46:16.226+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
[2024-05-06T07:46:16.243+0000] {process_utils.py:132} INFO - Sending 15 to group 682. PIDs of all processes in the group: [682]
[2024-05-06T07:46:16.247+0000] {process_utils.py:87} INFO - Sending the signal 15 to group 682
[2024-05-06T07:46:16.257+0000] {taskinstance.py:2607} ERROR - Received SIGTERM. Terminating subprocesses.
[2024-05-06T07:46:16.276+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2024-05-06T07:46:16.326+0000] {taskinstance.py:2890} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 465, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 432, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 400, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 235, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 252, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/scripts/extract_transform.py", line 50, in roberta_classifier
    output = model(**encoded_text)
             ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/torch/nn/modules/module.py", line 1532, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/torch/nn/modules/module.py", line 1541, in _call_impl
    return forward_call(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/transformers/models/roberta/modeling_roberta.py", line 1191, in forward
    outputs = self.roberta(
              ^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/torch/nn/modules/module.py", line 1532, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/torch/nn/modules/module.py", line 1541, in _call_impl
    return forward_call(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/transformers/models/roberta/modeling_roberta.py", line 828, in forward
    encoder_outputs = self.encoder(
                      ^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/torch/nn/modules/module.py", line 1532, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/torch/nn/modules/module.py", line 1541, in _call_impl
    return forward_call(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/transformers/models/roberta/modeling_roberta.py", line 517, in forward
    layer_outputs = layer_module(
                    ^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/torch/nn/modules/module.py", line 1532, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/torch/nn/modules/module.py", line 1541, in _call_impl
    return forward_call(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/transformers/models/roberta/modeling_roberta.py", line 406, in forward
    self_attention_outputs = self.attention(
                             ^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/torch/nn/modules/module.py", line 1532, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/torch/nn/modules/module.py", line 1541, in _call_impl
    return forward_call(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/transformers/models/roberta/modeling_roberta.py", line 333, in forward
    self_outputs = self.self(
                   ^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/torch/nn/modules/module.py", line 1532, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/torch/nn/modules/module.py", line 1541, in _call_impl
    return forward_call(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/transformers/models/roberta/modeling_roberta.py", line 259, in forward
    attention_probs = nn.functional.softmax(attention_scores, dim=-1)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/torch/nn/functional.py", line 1885, in softmax
    ret = input.softmax(dim)
          ^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 2609, in signal_handler
    raise AirflowTaskTerminated("Task received SIGTERM signal")
airflow.exceptions.AirflowTaskTerminated: Task received SIGTERM signal
[2024-05-06T07:46:16.368+0000] {taskinstance.py:1205} INFO - Marking task as UP_FOR_RETRY. dag_id=tweets_dag_v4, task_id=execute_python_script, execution_date=20240430T000000, start_date=20240506T074100, end_date=20240506T074616
[2024-05-06T07:46:16.487+0000] {standard_task_runner.py:110} ERROR - Failed to execute job 159 for task execute_python_script ((psycopg2.errors.ForeignKeyViolation) insert or update on table "task_fail" violates foreign key constraint "task_fail_ti_fkey"
DETAIL:  Key (dag_id, task_id, run_id, map_index)=(tweets_dag_v4, execute_python_script, scheduled__2024-04-30T00:00:00+00:00, -1) is not present in table "task_instance".

[SQL: INSERT INTO task_fail (task_id, dag_id, run_id, map_index, start_date, end_date, duration) VALUES (%(task_id)s, %(dag_id)s, %(run_id)s, %(map_index)s, %(start_date)s, %(end_date)s, %(duration)s) RETURNING task_fail.id]
[parameters: {'task_id': 'execute_python_script', 'dag_id': 'tweets_dag_v4', 'run_id': 'scheduled__2024-04-30T00:00:00+00:00', 'map_index': -1, 'start_date': datetime.datetime(2024, 5, 6, 7, 41, 0, 119835, tzinfo=Timezone('UTC')), 'end_date': datetime.datetime(2024, 5, 6, 7, 46, 16, 353086, tzinfo=Timezone('UTC')), 'duration': 316}]
(Background on this error at: https://sqlalche.me/e/14/gkpj); 682)
[2024-05-06T07:46:16.665+0000] {process_utils.py:80} INFO - Process psutil.Process(pid=682, status='terminated', exitcode=1, started='07:40:59') (682) terminated with exit code 1
